### ISSUE TITLE
Issues with ENV-variables override

### ISSUE THREAD

**Issue by sreichel** [OpenMage/magento-lts#4257]
Also did some tests ...

- not working if DB value exists
- Mage::getStoreConfig() does not work
- store config with overriden values should not have set "use default"
- store config with overriden values should be locked (to make it impossible to override config with ENV var)
- test fail for live data (rewritten tests to use installed OM)

_Originally posted by @sreichel in https://github.com/OpenMage/magento-lts/issues/3863#issuecomment-2398464478_

**Comment by sreichel** [comment]
@pquerner hey. Had you some time to review this?

**Comment by pquerner** [comment]
No. I'm currently loaded with other projects. Its best if the feature is disabled for now. (Its in a helper, and can be early returned)

I am thinking that I will have some time over the coming holidays / new year and will get back to you asap.

**Comment by sreichel** [comment]
Best X-mas wishes :christmas_tree:  :)

**Comment by pquerner** [comment]
What do you think about Reflectionclass? I found its the quickest and easiest way. And once the cache is warm, this shouldnt be an issue anymore. (Because Reflectionclass can be a little slow)

**Comment by pquerner** [comment]
WIP: https://github.com/pquerner/magento-lts/commit/6da31284621595880bad0d1baf2dec4e892cdcf0

**Comment by sreichel** [comment]
If you use DDEV, please use XHGUI for performance tests.

If it hurts and this feature is not meant to be used in prodution, we can wrap it into `isDeveloperMode()` 

Can you please review the admin-config section changes in https://github.com/OpenMage/magento-lts/pull/4261 to disable to save config, when ENV-vars are set.

**Comment by sreichel** [comment]
Can you please add a PR?
